satpy/tests/modifier_tests/test_crefl.py

```diff
@@ -349,14 +349,12 @@ def _start_dem_mock(self, tmpdir, url):
         rmock = rmock_obj.start()
         dem_fn = str(tmpdir.join(url))
         rmock.return_value = dem_fn
-        from netCDF4 import Dataset
-
-        nc = Dataset(dem_fn, 'w')
-        nc.createDimension('y', 10)
-        nc.createDimension('x', 10)
-        dem_var = nc.createVariable('averaged elevation', np.float32,
-                                    ('y', 'x'))
-        dem_var[:] = 0
+        from pyhdf.SD import SD, SDC
+
+        h = SD(dem_fn, SDC.WRITE | SDC.CREATE)
+        dem_var = h.create("averaged elevation", SDC.FLOAT32, (10, 10))
+        dem_var.setfillvalue(-999.0)
+        dem_var[:] = np.zeros((10, 10), dtype=np.float32)
```

Comment:
why not test both pyhdf and netcdf4 since you had the code for netcdf4?

Also: `tmpdir` is probably pointing to a tmpfs volume on linux, so file writing/reading will be fast, but it might not be the case on other systems. Maybe we should think in the future to keep this in memory only using `StringIO` or `BytesIO` (if at all possible)...

Reply:
The actual file is an HDF4 file. In the past I was using NetCDF4 built with HDF4 support to read these files. I cheated (poor decision) in the tests by just creating a NetCDF file. In writing these benchmarks I found out that the PyPI version of netcdf4-python does not include HDF4 support so I needed to switch to using pyhdf in the crefl code. Pyhdf can't read NetCDF files so this test needed to be updated to write an HDF4 file. This behavior matches the real-world situation where these DEM files are HDF4.